schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -187,46 +190,224 @@
+                        "isEnabled": {
+                          "enum": [
+                            "true",
+                            "false"
+                          ]
+                        },
+                        "scanType": {
+                          "type": "string"
+                        },
+                        "day": {
+                          "type": "string"
+                        },
+                        "time": {
+                          "type": "string"
+                        }
+                      },
+                      "required": [
+                        "isEnabled",
+                        "scanType",
+                        "day",
+                        "time"
+                      ]
+                    }
+                  },
+                  "required": [
+                    "AntimalwareEnabled",
+                    "Exclusions",
+                    "RealtimeProtectionEnabled",
+                    "ScheduledScanSettings"
+                  ]
+                }
+              },
+              "required": [
+                "publisher",
+                "type",
+                "typeHandlerVersion",
+                "autoUpgradeMinorVersion",
+                "settings"
+              ]
             },
-            "settings": {
-              "oneOf": [
-                { "type": "object" },
-                { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
-              ],
-              "description": "Microsoft.Compute/extensions - Settings"
+            {
+              "type": "object",
+              "properties": {
+                "publisher": {
+                  "enum": [
+                    "Microsoft.Compute"
+                  ]
+                },
+                "type": {
+                  "enum": [
+                    "CustomScriptExtension"
+                  ]
+                },
+                "typeHandlerVersion": {
+                  "type": "string"
+                },
+                "autoUpgradeMinorVersion": {
+                  "type": "boolean"
+                },
+                "settings": {
+                  "type": "object",
+                  "properties": {
+                    "fileUris": {
+                      "type": "array",
+                      "items": {
+                        "type": "string"
+                      }
+                    },
+                    "commandToExecute": {
+                      "type": "string"
+                    }
+                  },
+                  "required": [
+                    "fileUris",
+                    "commandToExecute"
+                  ]
+                }
+              },
+              "required": [
+                "publisher",
+                "type",
+                "typeHandlerVersion",
+                "autoUpgradeMinorVersion",
+                "settings"
+              ]
```

Comment:
I think this is the best we can do. Since publisher, type, and settings are different for each extension we can't really combine them and still make sure each extension follows their specific publisher, type, and settings properties.